packages/vat-data/test/test-virtual-classes.js

```diff
@@ -0,0 +1,90 @@
+// @ts-check
+
+// Modeled on test-heap-classes.js
+
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { M } from '@agoric/store';
+import {
+  defineVirtualFarClass,
+  defineVirtualFarClassKit,
+} from '../src/far-class-utils.js';
+
+const UpCounterI = M.interface('UpCounter', {
+  incr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+const DownCounterI = M.interface('DownCounter', {
+  decr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+test('test defineVirtualFarClass', t => {
+  const makeUpCounter = defineVirtualFarClass(
+    'UpCounter',
+    UpCounterI,
+    (x = 0) => ({ x }),
+    {
+      incr(y = 1) {
+        // @ts-expect-error TS doesn't know that `this` is a `Context`
+        const { state } = this;
+        state.x += y;
+        return state.x;
+      },
+    },
+  );
+  const upCounter = makeUpCounter(3);
+  t.is(upCounter.incr(5), 8);
+  t.is(upCounter.incr(1), 9);
+  t.throws(() => upCounter.incr(-3), {
+    message: 'In "incr" method of (UpCounter) arg 0: -3 - Must be >= 0',
+  });
+  // @ts-expect-error TS doesn't know that `this` is a `Context`
```

Comment:
There's no `this` in the next line so this is the wrong error message?

Reply:
Good catch! There were several of these, all of which are now changed to
```js
  // @ts-expect-error the type violation is what we're testing
```